### PR TITLE
Add subscriptions

### LINKED ISSUE #1338: Add support for Subscriptions
Add support for [http://dev.commercetools.com/http-api-projects-subscriptions.html](url) to our SDK. Biggest issue is to have some integration tests. I will first start without providing correct MQ destinations and just test that our resource representations are correct. Afterwards I will try to write an integration test with IronMQ or AWS. The biggest challenge here is to integrate it into our travis build.

### PR DESCRIPTION
**Pull request by katmatt**
Closes #1338

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/subscriptions/MessageSubscription.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+@JsonDeserialize(as = MessageSubscriptionImpl.class)
+@ResourceValue
+public interface MessageSubscription {
+    String getResourceTypeId();
+
+    @Nullable
+    List<String> getTypes();
+
+    static MessageSubscription of(final String resourceTypeId, final String... types) {
```
Comment (lauraluiz) [inline commercetools-models/src/main/java/io/sphere/sdk/subscriptions/MessageSubscription.java]: I would avoid using varargs or I would have it as syntactic sugar for `of(final String resourceTypeId, final List<String> types)`. Having it as the only method forces developers to use ugly arrays if they are providing types from a list.

File: commercetools-models/src/test/java/io/sphere/sdk/subscriptions/ChangeSubscriptionTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+        assertThat(jsonNode.get("resourceTypeId").asText()).isEqualTo(Review.resourceTypeId());
+    }
+
+    @Test
+    public void deserialize() throws Exception {
+        ChangeSubscription changeSubscription = SphereJsonUtils.readObject("{\"resourceTypeId\":\"category\"}", ChangeSubscription.class);
+        assertThat(changeSubscription.getResourceTypeId()).isEqualTo(Category.resourceTypeId());
+    }
+}
```
Comment (lauraluiz) [inline commercetools-models/src/test/java/io/sphere/sdk/subscriptions/ChangeSubscriptionTest.java]: I like UT! :D

File: commercetools-models/src/test/java/io/sphere/sdk/subscriptions/SnsUtils.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ */
+public class SnsUtils {
+    private SnsUtils() {
+    }
+
+    /**
+     * Creates a new sns test topic.
+     *
+     * @param snsClient the sqs client
+     * @return the arn of the created sns topic
```
Comment (lauraluiz) [inline commercetools-models/src/test/java/io/sphere/sdk/subscriptions/SnsUtils.java]: It took me a while to realize that ARN was an acronym for some AWS stuff. Could we write it uppercase in the Javadoc to leave it clear it's not a word? Same with SQS and SNS.

Comment (yanns) [inline commercetools-models/src/test/java/io/sphere/sdk/subscriptions/SnsUtils.java]: +1

Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/subscriptions/SnsUtils.java]: That's a good idea 👍 For me it was just another crazy acronym ;-)

File: commercetools-models/src/main/java/io/sphere/sdk/subscriptions/MessageSubscription.java (hunk 1 above)
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/subscriptions/MessageSubscription.java]: Good point and I agree with you, it's better to use lists here 😄

**Pull request merged**